ProjectMars/StepDefinitions/LanguageFeatureStepDefinitions.cs

```diff
@@ -0,0 +1,175 @@
+            Assert.That(p2 == actualResult1, "Language has not been added Successfully");
+        }
+        [When(@"I add new a '([^']*)' and '([^']*)'")]
+
+        public void WhenIAddNewAAnd(string p0, string p1)
+        {
+            profilePageObj.ResetLanguageRow(driver);
+            profilePageObj.CreateLanguageRecord(driver, p0, p1);
+        }
+
+
+        [When(@"I add an existing language'([^']*)' and '([^']*)'")]
+        public void WhenIAddAnExistingLanguageAnd(string p2, string p3)
+        {
+
+            profilePageObj.CreateLanguageRecord(driver, p2, p3);
+        }
+
+        [Then(@"the record should not be added '([^']*)'")]
+        public void ThenTheRecordShouldNotBeAdded(string p4)
+        {
+            string actualresult2 = driver.FindElement(By.XPath("//*[@class='ns-box-inner']")).Text;
+            Thread.Sleep(2000);
+            Assert.That(p4 == actualresult2, "Language has not been added Successfully");
+
+        }
+
+
+
+
+
+        //[Then(@"the record should  not be added as '([^']*)' and valid '([^']*)'")]
+        //public void ThenTheRecordShouldNotBeAddedAsAndValid(string p0, string p1)
+        //{
+
+
+        //       // Assert.That(p0 == "Please enter language and level" && p1 == "Fluent", " Language has not been added Successfully");
+        //}
+
+
+        [Given(@"I create  '([^']*)' and '([^']*)'")]
+        public void GivenICreateAnd(string p0, string p1)
+        {
+            profilePageObj.ResetLanguageRow(driver);
+            profilePageObj.CreateLanguageRecord(driver, p0, p1);
+        }
+
+        [When(@"I update '([^']*)' and '([^']*)'")]
+        public void WhenIUpdateAnd(string p2, string p3)
+        {
+            profilePageObj.EditLanguageRecord(driver, p2, p3);
+        }
+
+        [Then(@"the record should be updated as new'([^']*)' and '([^']*)'")]
+        public void ThenTheRecordShouldBeUpdatedAsNewAnd(string p2, string p3)
+        {
+            string newEditLanguage = profilePageObj.GetEditLanguage(driver);
+            string newEditLevel = profilePageObj.GetEditLevel(driver);
+            Assert.That(newEditLanguage == p2, "Language has not been updated sucessfully.");
+            Assert.That(newEditLevel == p3, "Level has not updated sucessfully");
+        }
+
+    
+        [Given(@"I create a '([^']*)' and '([^']*)'")]
+        public void GivenICreateAAnd(string p0, string p1)
+        {
+            profilePageObj.ResetLanguageRow(driver);
+            profilePageObj.CreateLanguageRecord(driver, p0, p1);
+        }
+
+     
+        [When(@"I delete a language")]
+        public void WhenIDeleteALanguage()
+        {
+          
+            profilePageObj.DeleteLanguageRecord(driver);
+        }
+
+        [Then(@"the '([^']*)'  should be deleted sucessfully")]
+        public void ThenTheShouldBeDeletedSucessfully(string p0)
+        {
+            string actualResult = driver.FindElement(By.XPath("//*[@class='ns-box-inner']")).Text;
```

Comment:
please move this within the page, create a method that returns the value and then use it! best practice to contain page logic within pages